sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -221,7 +224,7 @@ def _create_pipeline(self, credential, **kwargs):
         # type: (Any, **Any) -> Tuple[Configuration, Pipeline]
         self._credential_policy = None
         if hasattr(credential, "get_token"):
-            self._credential_policy = StorageBearerTokenCredentialPolicy(credential)
+            self._credential_policy = StorageBearerTokenCredentialPolicy(credential=credential, audience=kwargs["audience"])
```

Comment:
Does this work if kwargs does not have an audience listed or does it throw?

Reply:
That's a good point. So if I take the feedback that `StorageBearerTokenCredentialPolicy`'s `audience` is now a required str param to match `BearerTokenCredentialPolicy`, then it cannot be `None`. So I think the best pattern is:

1. In `_blob_client.py` we take in `audience` as the audience object if they want to use it.
2. In `base_client.py` we check if they provided one, if so -> use it. Else, we set audience to the default value.
3. That way, we can now send that value along to `_create_pipeline` and when it is pulled out is guaranteed to be a string (either the user-provided or the default).

Reply:
I think this design is fine, but you could also get rid of the `else` in `__init__` and then here just do `audience=kwargs.pop('audience', STORAGE_OAUTH_SCOPE)`.

Another option is to just do everything in this function and don't deal with anything in `__init__`.
```py
audience = kwargs.pop('audience', STORAGE_OAUTH_SCOPE)
self._credential_policy = StorageBearerTokenCredentialPolicy(credential=credential, audience=str(audience))
```

A little weird since you may call `str` on a string but I think that works fine. You can also just add the same `if` here if you don't like to stringing and string.